openstudiocore/src/osversion/VersionTranslator.cpp

```diff
@@ -3004,9 +3004,36 @@ std::string VersionTranslator::update_1_10_0_to_1_10_1(const IdfFile& idf_1_10_0
   IdfFile targetIdf(idd_1_10_1.iddFile());
   ss << targetIdf.versionObject().get();
 
+  auto zones = idf_1_10_0.getObjectsByType(idf_1_10_0.iddFile().getObject("OS:ThermalZone").get());
+
   for (const IdfObject& object : idf_1_10_0.objects()) {
     auto iddname = object.iddObject().name();
-    if (iddname == "OS:Sizing:Zone") {
+
+    if (iddname == "OS:ThermostatSetpoint:DualSetpoint") {
+      // Get all of the zones that point to this thermostat
+      std::vector<IdfObject> referencingZones;
+      for( const auto & zone : zones ) {
+        if( auto thermostateHandle = zone.getString(19) ) {
+          if( auto thermostat = idf_1_10_0.getObject(toUUID(thermostateHandle.get())) ) {
+            if( thermostat.get() == object ) {
+              referencingZones.push_back(zone);
+            }
+          }
+        }
+      }
+
+      // Clone the thermostat for every zone after the first
+      if( referencingZones.size() > 1u ) {
+        for( auto it = (referencingZones.begin() + 1); it != referencingZones.end(); ++it ) {
+          auto newThermostat = object.clone();
```

Comment:
Well magically the IdfFile / Workspace gears are appending a integer to guarantee uniqueness. (I just confirmed) But perhaps you would like to see the thermostat name derive from the zone name? I can be on board with that. Honestly I hadn't considered that since the name isn't visible in that app.